library/src/main/kotlin/build/buf/connect/Call.kt

```diff
@@ -0,0 +1,41 @@
+package build.buf.connect
+
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.atomic.AtomicReference
+
+/**
+ * A [Call] contains the way to make a blocking RPC call and cancelling the RPC.
+ */
+class Call<Output> {
+    private var executable: ((ResponseMessage<Output>) -> Unit) -> Unit = { }
+    private var cancel: () -> Unit = { }
+
+    /**
+     * Execute the underlying request.
+     */
+    fun execute(): ResponseMessage<Output> {
```

Comment:
What prevents this from being called multiple times?

Reply:
This isn't prevented.  OkHttp also does this!